lib_brewing/data/gm4_brewing-1.0/functions/check_brew.mcfunction

```diff
@@ -0,0 +1,6 @@
+# @s = none
+# run from process
+
+execute as @e[type=marker,tag=gm4_brewing_tick] at @s if block ~ ~ ~ brewing_stand{BrewTime:1s} run function gm4_brewing-1.0:prep_finish_brew
+
+execute if entity @e[type=marker,tag=gm4_brewing_tick,limit=1] run schedule function gm4_brewing-1.0:check_brew 1t
```

Comment:
We could aviod this selector by slightly modifying the function structure and by introducing another fake player.

By setting a fake player to a fixed value `A` at the start of this function and by setting that fake player to another fixed value `B` within `gm4_brewing-1.0:prep_finish_brew`, you could deduct that such a marker still exists. This would obviously require moving the block check into `gm4_brewing-1.0:prep_finish_brew` or a helper function which acts as a middle-man between this function and `gm4_brewing-1.0:prep_finish_brew`.

I have a suspicion that this decrease the average runtime in worlds with many entities.

Reply:
I've made a change, I don't understand the moving the block check part tho.

Reply:
Not moving the block check will alter the logic, here are two diagrams for you.

Behavior **before** 8ccb2d3:
| marker exists? | brewing_stand{BrewTime:1s}? | Outcome       |
|----------------|-----------------------------|---------------|
| yes            | yes                         | Abort loop    |
| yes            | no                          | Continue loop |
| no             | yes                         | Abort loop    |
| no             | no                          | Abort loop    |

Behavior **after** 8ccb2d3:
| marker exists? | brewing_stand{BrewTime:1s}? | Outcome       |
|----------------|-----------------------------|---------------|
| yes            | yes                         | Abort loop    |
| yes            | no                          | Continue loop |
| no             | yes                         | Continue loop |
| no             | no                          | Continue loop |

Reply:
But the score is only ever set to 0 if it exists. This would indeed continue the loop if the marker gets unloaded within the 16 ticks, but will also allow it to catch it when it does get reloaded. TBH I think this or the previous version is fine; for the previous version, the loop only ever runs if there is a brewing stand within 16 ticks of finishing the brew

Reply:
I see, that's alright. Especially as the loop would be escaped if another brewing stand finishes brewing, so infinite loops would be unlikely.
@misode what do you think?